lib/jekyll-last-modified-at/git.rb

```diff
@@ -33,6 +35,80 @@ def git_repo?
                          false
         end
       end
+
+      def last_commit_date(path, use_git_cache = false) # rubocop:disable Style/OptionalBooleanParameter
+        if use_git_cache
+          build_cache if @lcd_cache.empty?
+          @lcd_cache[path]
+        else
+          Executor.sh(
+            'git',
+            '--git-dir',
+            top_level_directory,
+            'log',
+            '-n',
+            '1',
+            '--format="%ct"',
+            '--',
+            path
+          )[/\d+/]
+        end
+      end
+
+      def first_commit_date(path, use_git_cache = false) # rubocop:disable Style/OptionalBooleanParameter
+        if use_git_cache
+          build_cache if @lce_cache.empty?
+          @lce_cache[path]
+        else
+          Executor.sh(
+            'git',
+            '--git-dir',
+            top_level_directory,
+            'log',
+            '--format="%ct"',
+            '--',
+            path
+          ).split("\n")[-1][/\d+/]
+        end
```

Comment:
Some issues with the command in the PR:
- doesn't work with renames. `--follow` does
- `--diff-filter=A` will massively reduce the number of lines to process
- %ct doesn't account for overwrites. %aD does

Git command that accounts for all the above, as given in #90:

`git log --diff-filter=A --follow --format=%aD -1 -- <fname>`

Reply:
`ct` vs `at` raises a good question on how to handle people who rewrite history.

In my experience, `--amend`s usually happen locally, so at t=`%at` the file usually hasn't been published yet. There's also no guarantee that the file existed at t=`at` because it might have been added in the amend, for example because the author did `commit -a` (necessitating a `git add` and `commit --amend` to actually add the new file). So, while there are certainly cases of people force-pushing amends, I still feel `ct` is more likely to be the correct time in most cases. Thoughts?

Reply:
Agree, all good points for favoring ct.

Reply:
Great! And the other changes have been implemented, thanks for suggesting them. Enjoy the fork! 🙂 🍽️